### PR TITLE
Auto-generate and verify mock clients

### PR DESCRIPTION
**Pull request by chetanugale**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
- Script based mock generate and update with Makefile and mockgen
- Updated mocks for _./pkg/azureclients/armclient/interface.go_ for testing scenario.
- Script operational for single or multiple mock generations

**Which issue(s) this PR fixes**:

Fixes # https://github.com/kubernetes-sigs/cloud-provider-azure/issues/624

**Special notes for your reviewer**:
- Encountered some issues with go generate like already generated mocks were skipped at update scenario.

**Release note**:
```
none
```

### CODE REVIEW

**Review by andyzhangx — COMMENTED** [review]
/ok-to-test

File: hack/manage-mocks.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+COPYRIGHT_FILE=./hack/boilerplate/boilerplate.generatego.txt
+TARGET_DIR=./pkg/azureclients
+IFACE_FILE=interface.go
```
Comment (feiskyer) [inline hack/manage-mocks.sh:19]: Could you read these two params from environment varilable?

**Review by feiskyer — COMMENTED** [review]
thanks for the contribution. There is a little difference for generated mock clients, could you  fix the unit test failures?

File: hack/manage-mocks.sh (hunk 1 above)
Comment (chetanugale) [reply hack/manage-mocks.sh:19]: can you please point to any example which will help me resolve this comment.

Also Need help with unit test debugging...I checked the generated mock_client...only boilerplate part is updated..remaining file is same..pointers regarding the same will also be helpful

Comment (feiskyer) [reply hack/manage-mocks.sh]: Thanks for working on this. I pushed the required changes to get this merged sooner.

**Pull request merged**